test/pkg/ptptesthelper/ptptesthelper.go

```diff
@@ -272,53 +272,149 @@ func CheckSlaveSyncWithMaster(fullConfig testconfig.TestConfig) {
 }
 
-func RebootSlaveNode(fullConfig testconfig.TestConfig) {
-	logrus.Info("Rebooting system starts ..............")
+// To delete a ptp test priviledged daemonset
+func DeletePtpTestPrivilegedDaemonSet(daemonsetName, daemonsetNamespace string) {
+	k8sPriviledgedDs.SetDaemonSetClient(client.Client.Interface)
+	err := k8sPriviledgedDs.DeleteDaemonSet(daemonsetName, daemonsetNamespace)
+	if err != nil {
+		logrus.Errorf("error deleting %s daemonset, err=%s", daemonsetName, err)
+	}
+}
 
+// To create a ptp test privileged daemonset
+func CreatePtpTestPrivilegedDaemonSet(daemonsetName, daemonsetNamespace, daemonsetContainerName string) *corev1.PodList {
 	const (
 		imageWithVersion = "quay.io/testnetworkfunction/debug-partner:latest"
 	)
-
 	// Create the client of Priviledged Daemonset
 	k8sPriviledgedDs.SetDaemonSetClient(client.Client.Interface)
 	// 1. create a daemon set for the node reboot
 	dummyLabels := map[string]string{}
-	rebootDaemonSetRunningPods, err := k8sPriviledgedDs.CreateDaemonSet(pkg.RebootDaemonSetName, pkg.RebootDaemonSetNamespace, pkg.RebootDaemonSetContainerName, imageWithVersion, dummyLabels, pkg.TimeoutIn5Minutes)
+	daemonSetRunningPods, err := k8sPriviledgedDs.CreateDaemonSet(daemonsetName, daemonsetNamespace, daemonsetContainerName, imageWithVersion, dummyLabels, pkg.TimeoutIn5Minutes)
+
 	if err != nil {
 		logrus.Errorf("error : +%v\n", err.Error())
 	}
-	nodeToPodMapping := make(map[string]corev1.Pod)
-	for _, dsPod := range rebootDaemonSetRunningPods.Items {
-		nodeToPodMapping[dsPod.Spec.NodeName] = dsPod
+	return daemonSetRunningPods
+}
+
+func RecoverySlaveNetworkOutage(fullConfig testconfig.TestConfig, skippedInterfaces map[string]bool) {
+	logrus.Info("Recovery network outage begins ...........")
+
+	// Get a slave pod
+	slavePod, err := ptphelper.GetPTPPodWithPTPConfig((*ptpv1.PtpConfig)(fullConfig.DiscoveredClockUnderTestPtpConfig))
+	if err != nil {
+		logrus.Error("Could not determine ptp daemon pod selected by ptpconfig")
+	}
+	// Get the slave pod's node name
+	slavePodNodeName := slavePod.Spec.NodeName
+	logrus.Info("slave node name is ", slavePodNodeName)
+
+	// Get the pod from ptp test daemonset set on the slave node
+	networkRecoveryDaemonSetRunningPods := CreatePtpTestPrivilegedDaemonSet(pkg.RecoveryNetworkOutageDaemonSetName, pkg.RecoveryNetworkOutageDaemonSetNamespace, pkg.RecoveryNetworkOutageDaemonSetContainerName)
+	var networkOutageRecoveryDaemonsetPod corev1.Pod
+
+	Expect(len(networkRecoveryDaemonSetRunningPods.Items)).To(BeNumerically(">", 0), "no damonset pods found in the namespace "+pkg.RecoveryNetworkOutageDaemonSetNamespace)
+	for _, dsPod := range networkRecoveryDaemonSetRunningPods.Items {
+		if dsPod.Spec.NodeName == slavePodNodeName {
+			networkOutageRecoveryDaemonsetPod = dsPod
+			break
+		}
+	}
+	logrus.Info("outage recovery pod name is ", networkOutageRecoveryDaemonsetPod.Name)
```

Comment:
This could be nil if the NodeName doesnt match

Reply:
I think that would be highly unlikely, because the Daemonset pods would be running on each node and the code waited until all these pods are running fine to declare that daemonset is deployed successfully.

Reply:
I added a check here though.